lib/logstash/gemfile.rb

```diff
@@ -0,0 +1,173 @@
+module LogStash
+
+  class GemfileError < StandardError; end
+
+  class Gemfile
+    attr_accessor :gemset
+
+    HEADER = \
+      "# This is a Logstash generated Gemfile.\n" + \
+      "# If you modify this file manually all comments and formatting will be lost.\n\n"
+
+    # @params io [IO] any IO object that supports read, write, truncate, rewind
+    def initialize(io)
+      @io = io
+      @gemset = nil
+    end
+
+    def load
+      @gemset ||= Dsl.parse(@io.read)
+      self
+    end
+
+    def save
+      raise(GemfileError, "a Gemfile must first be loaded") unless @gemset
```

Comment:
sure. but again, this is internal code that we use just for the plugin tools and if you hit this exception that's because you did a really bad sequence in your code. in other words, this should really not happen.